### ISSUE TITLE
A weird sound shuffling bug at the end of track(any map, any track)

### ISSUE THREAD

**Issue by Xansaibot** [yquake2/xatrix#42]
Don't mind that i am not in the game(it's alt-tabed while this was recorded). Instead of playing on a infinite loop, the game begins to play different song for 1-3 seconds (from previous of current level).

[Weird sound shuffle bug_.zip](https://github.com/yquake2/xatrix/files/3892849/Weird.sound.shuffle.bug_.zip)

P.S. Latest Patch of Yamagi is installed, and this occurs only in The Reckoning.
Shuffle option is set to OFF.

Edit: attaching additional archive with video proof of a bug
[Weird sound shuffling bug 2.zip](https://github.com/yquake2/xatrix/files/3894088/Weird.sound.shuffling.bug.2.zip)

**Comment by Yamagi** [comment]
I hate the shuffle code. ;) The problem is, the original code didn't implement shuffling by itself. It just told the CD drive to play a random track. Hacking shuffling into the OGG playback stuff wasn't straight forward and the need for different track number <-> file name mapping (to support different  distributions of the game, for example the GOG.com version is different) didn't make things easier. 

We had some fixes to shuffling since the last release. Can you test with our latest testbuild? It can be found here: https://deponie.yamagi.org/quake2/windows/testbuilds/quake2-20191117-fbf103b-7.42-RC1.zip It'll become version 7.42.

**Comment by Xansaibot** [comment]
> I hate the shuffle code. ;) The problem is, the original code didn't implement shuffling by itself. It just told the CD drive to play a random track. Hacking shuffling into the OGG playback stuff wasn't straight forward and the need for different track number <-> file name mapping (to support different distributions of the game, for example the GOG.com version is different) didn't make things easier.
> 
> We had some fixes to shuffling since the last release. Can you test with our latest testbuild? It can be found here: https://deponie.yamagi.org/quake2/windows/testbuilds/quake2-20191117-fbf103b-7.42-RC1.zip It'll become version 7.42.

No, it still happens unfortunately:(

**Comment by Yamagi** [comment]
Hi,
I listened to the files I've ripped many years ago from the original disks (the release versions from) and they sound okay. Maybe you files are somehow corrupted, maybe by a problem with the cd extractor?


Regarding the music folders:

- For the gog.com version the music is placed in a directory called `music/` next to `baseq2` and so on. That's because GOG provides the music with their proprietary track mapping.
- For all other versions the music is placed at `baseq2/music`, `rogue/music` and `xatrix/music`. If an addon music dir is missing, but the baseq2 music dir is there, the game falls back to the later.

Have a look here for comprehensive file lists: https://github.com/yquake2/yquake2/blob/master/doc/090_filelists.md
